projects/rocfft/shared/fft_params.h

```diff
@@ -2540,6 +2540,134 @@ class fft_params
         len.insert(len.begin(), nbatch);
         distribute_field(gpusperrank, brick_grid, ofields, len, num_ranks);
     }
+
+    // Apply load operations specified by this struct to the host-side
+    // input before we pass it to the reference FFT
+    void apply_host_load_ops(std::vector<hostbuf>& input) const
+    {
+        // Currently no load operations can be specified
+    }
+
+    // Apply store operations specified by this struct to the host-side
+    // output after we get it from the reference FFT
+    void apply_host_store_ops(std::vector<hostbuf>& output) const
+    {
+        // Don't bother iterating over the data if we don't have to
+        if(scale_factor == 1.0)
+            return;
+
+        switch(otype)
+        {
+            // Reference FFT always works with complex-interleaved data even if
+            // params specifies planar
+        case fft_array_type_complex_interleaved:
+        case fft_array_type_hermitian_interleaved:
+        case fft_array_type_complex_planar:
+        case fft_array_type_hermitian_planar:
+        {
+            switch(precision)
+            {
+            case fft_precision_half:
+            {
+                const size_t elem_size = sizeof(rocfft_complex<rocfft_fp16>);
+                const size_t num_elems = output.front().size() / elem_size;
+
+                auto output_begin
+                    = reinterpret_cast<rocfft_complex<rocfft_fp16>*>(output.front().data());
+                for(size_t i = 0; i < num_elems; ++i)
+                {
+                    auto& element = output_begin[i];
+                    if(scale_factor != 1.0)
+                        element = element * scale_factor;
```

Comment:
```suggestion
                    output_begin[i] *= scale_factor;
```
here and in subsequent cases, given that `scale_factor == 1.0` triggers an early return anyways.